crates/jstz_api/src/stream/readable/underlying_source.rs

```diff
@@ -0,0 +1,408 @@
+    pub cancel: Option<UnderlyingSourceCancelCallback>,
+
+    /// [Streams Standard - § 4.2.3.][https://streams.spec.whatwg.org/#dom-underlyingsource-type]
+    /// > **`type` (byte streams only), of type ReadableStreamType**
+    /// >
+    /// > Can be set to "bytes" to signal that the constructed ReadableStream is a readable byte stream. This ensures that the resulting ReadableStream will successfully be able to vend BYOB readers via its getReader() method. It also affects the controller argument passed to the start() and pull() methods; see below.
+    /// >
+    /// > For an example of how to set up a readable byte stream, including using the different controller interface, see § 10.3 A readable byte stream with an underlying push source (no backpressure support).
+    /// >
+    /// > Setting any value other than "bytes" or undefined will cause the ReadableStream() constructor to throw an exception.
+    pub r#type: Option<ReadableStreamType>,
+
+    /// [Streams Standard - § 4.2.3.][https://streams.spec.whatwg.org/#dom-underlyingsource-autoallocatechunksize]
+    /// > **`autoAllocateChunkSize` (byte streams only), of type unsigned long long**
+    /// >
+    /// > Can be set to a positive integer to cause the implementation to automatically allocate buffers for the underlying source code to write into. In this case, when a consumer is using a default reader, the stream implementation will automatically allocate an ArrayBuffer of the given size, so that controller.byobRequest is always present, as if the consumer was using a BYOB reader.
+    /// >
+    /// > This is generally used to cut down on the amount of code needed to handle consumers that use default readers, as can be seen by comparing § 10.3 A readable byte stream with an underlying push source (no backpressure support) without auto-allocation to § 10.5 A readable byte stream with an underlying pull source with auto-allocation.
+    pub auto_allocate_chunk_size: Option<idl::UnsignedLongLong>, // TODO [EnforceRange]
+}
+
+impl Finalize for UnderlyingSource {
+    fn finalize(&self) {}
+}
+
+unsafe impl Trace for UnderlyingSource {
+    custom_trace!(this, {
+        mark(&this.this);
+        mark(&this.start);
+        mark(&this.pull);
+        mark(&this.cancel);
+    });
+}
+
+// TODO derive this implementation with a macro?
+impl TryFromJs for UnderlyingSource {
+    fn try_from_js(value: &JsValue, context: &mut Context<'_>) -> JsResult<Self> {
+        // TODO check that this function works as intended in all cases,
+        // and move it either to a new derive macro for TryFromJs, or to JsObject
+        #[allow(non_snake_case)]
+        pub fn get_JsObject_property(
```

Comment:
Why non-snake-case? I'm not too sure why the deviation from standard style is helpful here

Reply:
What would you prefer? `get_jsobject_property`?